TombEngine/Objects/Utils/object_helper.cpp

```diff
@@ -9,6 +9,50 @@
 #include "Objects/TR5/Object/tr5_pushableblock.h"
 #include "Specific/level.h"
 
+
+void AssignObjectMeshSwap(ObjectInfo* obj, int requiredMeshSwap, const std::string& baseName, const std::string& requiredName)
+{
+	if (Objects[requiredMeshSwap].loaded)
+		obj->meshSwapSlot = requiredMeshSwap;
+	else
+		TENLog("The slot: " + requiredName + " is not loaded, " + baseName + " will have problem to swapmesh (could cause problem with entity not working correctly).", LogLevel::Warning);
+}
+
+void CheckIfSlotExist(int requiredObj, const std::string& baseName, const std::string& requiredName)
+{
+	if (!Objects[requiredObj].loaded)
+		TENLog("The slot: " + requiredName + " is not loaded, " + baseName + " will have problem to work correctly (could cause crash).", LogLevel::Warning);
+}
+
+void AssignObjectAnimations(ObjectInfo* obj, int requiredObj, int requiredSecondObj, const std::string& baseName, const std::string& requiredName, const std::string& requiredSecondName)
```

Comment:
Please only check one slot requirement at a time, remove `SecondObj`. To solve cases when slot needs to check 2 requirements, you can return a bool from this function and only check the next one if first one failed.

Reply:
Fixed.